torch/fx/passes/dialect/aten/canonicalize.py

```diff
@@ -0,0 +1,43 @@
+from torch.fx.graph_module import GraphModule
+from torch._ops import OpOverload, OpOverloadPacket
+
+def is_canonical(gm: GraphModule) -> bool:
+
+    for node in gm.graph.nodes:
+        if node.op in {"placeholder", "output"}:
```

Comment:
Hmm another question - is it the case that we expect a canonical graph to always be a bunch of straight-line code, with a single output node at the very end? Maybe it's worth asserting that.

(Although I guess the only way that wouldn't be the case is if we have control flow directly in the IR, which we don't today)

Reply:
hmm... Does fx.Graph even allows multiple output nodes? I was under the impression that it's not even a valid graph. 

For control flow node, I would still expect both branches to converge into a single output.